File: 프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<int> people, int limit) {
    
    int answer = 0;
    
    sort(people.rbegin(), people.rend());
    
    int heavy=0, light=people.size()-1;
    
    while(heavy <= light){
        
        if (heavy == light && people[heavy] <= limit){
            // answer++;
            // heavy++;
        }else if (people[heavy] + people[light] <= limit){
        
            // answer++;
            // heavy++;
            light--;
            
        }else if(people[heavy] <= limit){
            // heavy++;
            // answer++;

        }    
        heavy++;
        answer++;
    }

    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort()
    left, right = 0, len(people)-1
    
    while left <= right:
        sum = people[left] + people[right]    
    
        if sum > limit:
            right -= 1      
        else:
            left +=1
            right -=1
        
        answer +=1
        
    return answer